src/com/cft/contactmerge/tests/CompareContactPartsTest.java

```diff
@@ -5,67 +5,302 @@
     }
 
     @Test
     void doNamesMatch_Yes_CaseDoesNotMatter() {
-        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("john", "DOE", "JOHN", "Doe"));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("john", "DOE", "JOHN", "Doe"), firstAndLastNameFailedMsg("john", "DOE", "JOHN", "Doe"));
     }
 
     @Test
     void doNamesMatch_Yes_IgnoreSpaces() {
-        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" John", "Doe", "John ", " Doe  "));
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" John", "Doe", "John ", " Doe  "), firstAndLastNameFailedMsg(" John", "Doe", "John ", " Doe  "));
+    }
+
+    @Test
+    void doNamesMatch_Yes_IgnoreSpaces_CaseInsensitive() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" JoHn", "DOE", "JOHN ", " Doe  "), firstAndLastNameFailedMsg(" JoHn", "DOE", "JOHN ", " Doe  "));
+    }
+
+    //------------------------------------------------------------------------- Punctuation First Names -------------------------------------------------------------------------
+    @Test
+    void doNamesMatch_Yes_IgnorePunctuationInFirstName_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("aN-denIse ", "Doe ", " An Denise ", "   doe "), firstAndLastNameFailedMsg("aN-denIse ", "Doe ", " An Denise ", "   doe "));
+    }
+
+    @Test
+    void doNamesMatch_Yes_IgnoreMultiPunctuationInFirstName_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("aN-De-denIse ", "dOe ", " An dE Denise ", "   doe "), firstAndLastNameFailedMsg("aN-De-denIse ", "dOe ", " An dE Denise ", "   doe "));
+    }
+
+    @Test
+    void doNamesMatch_No_IgnoreMultiPunctuationInFirstName_CaseInsensitive_IgnoreSpaces_DifferentLastNames() {
+        assertEquals(AnswerType.no, CompareContactParts.doNamesMatch(" aN-De-denIse", " dOe", "   An dE Denise  ", " adams  "), firstAndLastNameFailedMsg(" aN-De-denIse", " dOe", "   An dE Denise  ", " adams  "));
+    }
+
+    //-------------------------------------------------------------------------- Punctuation Last Names -------------------------------------------------------------------------
+    @Test
+    void doNamesMatch_Yes_IgnorePunctuationInLastName_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch("Jane ", "Doe-adams ", "Jane", "  Doe Adams"), firstAndLastNameFailedMsg("Jane ", "Doe-adams ", "Jane", "  Doe Adams"));
+    }
+
+    @Test
+    void doNamesMatch_Yes_IgnoreMultiPunctuationInLastName_CaseInsensitive_IgnoreSpaces() {
+        assertEquals(AnswerType.yes, CompareContactParts.doNamesMatch(" arEb", " Sa-LA-Democh", "Areb", "sa la democh  "), firstAndLastNameFailedMsg(" arEb", " Sa-LA-Democh", "Areb", "sa la democh  "));
```

Comment:
See above comments for this and the following test. They don't seem to add anything and are just repeating tests we have already done.